samples/video-call-core/app/src/main/java/com/cloudflare/videocall/sample/MainActivity.kt

```diff
@@ -0,0 +1,303 @@
+      ).show()/*
+         * Note: calling meeting.release is currently compulsory when closing the Activity after
+         * reconnection failure.
+         * */
+      meeting?.release(onSuccess = { finish() }, onFailure = { finish() })
+    }
+
+    private fun onReconnectedToMeetingRoom() {
+      Log.d(TAG, "onReconnectedToMeetingRoom")
+      Toast.makeText(applicationContext, "Connection restored", Toast.LENGTH_SHORT).show()
+    }
+
+    private fun onReconnectingToMeetingRoom() {
+      Log.d(TAG, "onReconnectingToMeetingRoom")
+      Toast.makeText(
+        applicationContext, "Connection lost. Trying to reconnect...", Toast.LENGTH_SHORT
+      ).show()
+      showProgressBar()
+      disableCallControls()
+    }
+  }
+
+  private val localUserEventsListener = object : RtkSelfEventListener {
+    override fun onVideoUpdate(isEnabled: Boolean) {
+      meeting?.localUser?.let { refreshLocalUserVideo(it) }
+      binding.buttonCameraToggle.isEnabled = true
+    }
+  }
+
+  private val remoteUserUpdateListener = object : RtkParticipantUpdateListener {
+    override fun onVideoUpdate(participant: RtkMeetingParticipant, isEnabled: Boolean) {
+      if (participant.id == remoteUser?.id) {
+        refreshRemoteUserVideo(participant as RtkRemoteParticipant)
+      }
+    }
+  }
+
+  private val participantsEventListener = object : RtkParticipantsEventListener {
+    override fun onParticipantJoin(participant: RtkRemoteParticipant) {
+      Log.d(TAG, "onParticipantJoin, ${participant.name}")
+    }
+
+    override fun onParticipantLeave(participant: RtkRemoteParticipant) {
+      Log.d(TAG, "onParticipantLeave, ${participant.name}")
+    }
+
+    override fun onActiveParticipantsChanged(active: List<RtkRemoteParticipant>) {
+      Log.d(TAG, "onActiveParticipantsChanged, ${active.map { it.name }}")/*
+         * Select the current remote user from active participants, ignoring any old instance
+         * if the user reconnected after a network issue.
+         * */
+      val remoteUser = active.firstOrNull { it.id != this@MainActivity.remoteUser?.id } ?: run {
+        Log.d(
+          TAG,
+          "onActiveParticipantsChanged, remote user is either already rendered or hasn't joined the call yet"
+        )
+        return
+      }
+      setUpRemoteUser(remoteUser)
+    }
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    _binding = ActivityMainBinding.inflate(layoutInflater)
+    setContentView(binding.root)
+    initCallControls()
+    initMeetingClient()
+  }
+
+  override fun onDestroy() {
+    remoteUser?.removeParticipantUpdateListener(remoteUserUpdateListener)
+    meeting?.removeMeetingRoomEventListener(meetingRoomEventsListener)
+    meeting?.removeSelfEventListener(localUserEventsListener)
+    meeting?.removeParticipantsEventListener(participantsEventListener)
+    _binding = null
+    super.onDestroy()
+  }
+
+  private fun initCallControls() {
```

Comment:
```suggestion
  private fun setupUI() {
```

Since we are not initialising any views now and just setting them up.